job/common/src/main/java/alluxio/job/wire/TaskInfo.java

```diff
@@ -137,18 +154,24 @@ public TaskInfo setErrorMessage(String errorMessage) {
    * @param result the result
    * @return the updated task info object
    */
-  public TaskInfo setResult(byte[] result) {
-    mResult = result == null ? null : Arrays.copyOf(result, result.length);
+  public TaskInfo setResult(Serializable result) {
+    mResult = result;
     return this;
   }
 
   /**
    * @return proto representation of the task info
    * @throws IOException if serialization fails
    */
-  public alluxio.grpc.TaskInfo toProto() throws IOException {
-    ByteBuffer result =
-        mResult == null ? null : ByteBuffer.wrap(SerializationUtils.serialize(mResult));
+  public alluxio.grpc.TaskInfo toProto() {
+    ByteBuffer result = null;
+    try {
+      result = mResult == null ? null : ByteBuffer.wrap(SerializationUtils.serialize(mResult));
+    } catch (IOException e) {
+      // TODO(bradley) better error handling
+      LOG.warn("Failed to serialize {} : {}", mResult, e.getMessage());
+      LOG.debug("Exception: ", e);
```

Comment:
I suggest using `LOG.isDebugEnabled` and only log one message, it is not guaranteed that these two messages will be printed together.

Reply:
This is actually a copy paste from an existing code in TaskExecutorManager.java